### PR TITLE
Add torch tracing unit test in test_attentions

### PR DESCRIPTION
**Pull request by mehdimashayekhi**
Added a unit test in tests_attentions.py for testing torch script ability of attentions. Note that, in the unit test I am using tracing , but the better approach would be torch scripting given that there are some control flows (i.e., if then) in the implementations, however, currently torch script does not support some functionalities such as having variable number of arguments (i.e, *args, **kwargs). 

CircleCI passes the unit tests.

### CODE REVIEW

**Review by dianaml0 — COMMENTED** [review]
Thanks @mehdimashayekhi ! This is great to have! Looks good, just had a few comments

File: tests/test_attentions.py [hunk 1]
```diff
@@ -243,4 +242,43 @@ def test_causal(
...
+        "favor",
+        "global",
+        "local",
+        "random",
+    }:
+        # pyre-fixme[29]: The library function `pytest.skip` is not supported by Pyre.
+        pytest.skip(f"{attention_name} does not support scripting yet.")
+    multi_head = _get_multihead(attention_name, 0.0, 0.0, False, heads, device)
+
+    seq_q = SEQ - 16
```
[the remark this answers is not in the document]
Comment (blefaudeux) [reply tests/test_attentions.py:263]: it was part of the unit test used as an example I think, that was to make sure Q,K,V could have different sizes. In this particular test it's probably not super useful but does not hurt ?

**Comment by blefaudeux** [comment]
Thanks @mehdimashayekhi for this great PR !

File: tests/test_attentions.py [hunk 2]
```diff
@@ -243,4 +242,43 @@ def test_causal(
     ), res_sum
 
 
+@pytest.mark.parametrize("heads", [2])
+@pytest.mark.parametrize("attention_name", ATTENTION_REGISTRY.keys())
+@pytest.mark.parametrize("device", DEVICES)
+def test_torch_script_ability(
```
[the remark this answers is not in the document]
Comment (mehdimashayekhi) [reply tests/test_attentions.py:249]: Parametrized it w/ dropout.

**Comment by dianaml0** [comment]
@mehdimashayekhi I think the gpu tests are failing with resource unavailable because this is a fork and so using your personal account

**Comment by dianaml0** [comment]
@mehdimashayekhi if you follow the project instead of the project fork in circleCI that should fix this: https://circleci.com/docs/2.0/oss/

**Review by dianaml0 — APPROVED** [review]
Thanks for the updates!

**Pull request merged**